// angular-8-datatable/src/app/datapaginator-component/datapaginator.component.ts
import { Component, Input, SimpleChange, OnChanges, Optional } from "@angular/core";
import { DataTableService, PageEvent } from "../services/datatable.service";

@Component({
    selector: "mfPaginator",
    templateUrl: './datapaginator.component.html'
})
export class DataTablePaginatorComponent implements OnChanges {

    @Input("mfTable") inputMfTable: DataTableService;

    private mfTable: DataTableService;

    public activePage: number;
    public rowsOnPage: number;
    public dataLength: number = 0;
    public lastPage: number;

    public constructor(@Optional() private injectMfTable: DataTableService) {
    }

    public ngOnChanges(changes: { [key: string]: SimpleChange }): any {
        this.mfTable = this.inputMfTable || this.injectMfTable;
        this.onPageChangeSubscriber(this.mfTable.getPage());
        this.mfTable.onPageChange.subscribe(this.onPageChangeSubscriber);
    }

    public setPage(pageNumber: number): void {
        this.mfTable.setPage(pageNumber, this.rowsOnPage);
    }

    public setRowsOnPage(rowsOnPage: number): void {
        this.mfTable.setPage(this.activePage, rowsOnPage);
    }

    private onPageChangeSubscriber = (event: PageEvent) => {
        this.activePage = event.activePage;
        this.rowsOnPage = event.rowsOnPage;
        this.dataLength = event.dataLength;
        this.lastPage = Math.ceil(this.dataLength / this.rowsOnPage);
    };
}
