// angular-8-datatable/src/dist/out-tsc/src/app/datapaginator-component/bootstrapdatapaginator.component.js
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import * as _ from 'lodash';
import { DataTableService } from '../services/datatable.service';
let BootstrapDataTablePaginator = class BootstrapDataTablePaginator {
    constructor() {
        this.rowsOnPageSet = [];
        this.minRowsOnPage = 0;
    }
    ngOnChanges(changes) {
        if (changes.rowsOnPageSet) {
            this.minRowsOnPage = _.min(this.rowsOnPageSet);
        }
    }
};
tslib_1.__decorate([
    Input("rowsOnPageSet"),
    tslib_1.__metadata("design:type", Object)
], BootstrapDataTablePaginator.prototype, "rowsOnPageSet", void 0);
tslib_1.__decorate([
    Input("mfTable"),
    tslib_1.__metadata("design:type", DataTableService)
], BootstrapDataTablePaginator.prototype, "mfTable", void 0);
BootstrapDataTablePaginator = tslib_1.__decorate([
    Component({
        selector: 'mfBootstrapPaginator',
        templateUrl: './bootstrapdatapaginator.component.html'
    })
], BootstrapDataTablePaginator);
export { BootstrapDataTablePaginator };
