// angular-8-datatable/src/dist/out-tsc/src/app/datapaginator-component/datapaginator.component.js
import * as tslib_1 from "tslib";
import { Component, Input, Optional } from "@angular/core";
import { DataTableService } from "../services/datatable.service";
let DataTablePaginatorComponent = class DataTablePaginatorComponent {
    constructor(injectMfTable) {
        this.injectMfTable = injectMfTable;
        this.dataLength = 0;
        this.onPageChangeSubscriber = (event) => {
            this.activePage = event.activePage;
            this.rowsOnPage = event.rowsOnPage;
            this.dataLength = event.dataLength;
            this.lastPage = Math.ceil(this.dataLength / this.rowsOnPage);
        };
    }
    ngOnChanges(changes) {
        this.mfTable = this.inputMfTable || this.injectMfTable;
        this.onPageChangeSubscriber(this.mfTable.getPage());
        this.mfTable.onPageChange.subscribe(this.onPageChangeSubscriber);
    }
    setPage(pageNumber) {
        this.mfTable.setPage(pageNumber, this.rowsOnPage);
    }
    setRowsOnPage(rowsOnPage) {
        this.mfTable.setPage(this.activePage, rowsOnPage);
    }
};
tslib_1.__decorate([
    Input("mfTable"),
    tslib_1.__metadata("design:type", DataTableService)
], DataTablePaginatorComponent.prototype, "inputMfTable", void 0);
DataTablePaginatorComponent = tslib_1.__decorate([
    Component({
        selector: "mfPaginator",
        templateUrl: './datapaginator.component.html'
    }),
    tslib_1.__param(0, Optional()),
    tslib_1.__metadata("design:paramtypes", [DataTableService])
], DataTablePaginatorComponent);
export { DataTablePaginatorComponent };
