// angular-8-datatable/src/dist/out-tsc/src/app/datasorter-component/datasorter.component.js
import * as tslib_1 from "tslib";
import { Component, Input } from "@angular/core";
import { DataTableService } from "../services/datatable.service";
let DataTableSorterComponent = class DataTableSorterComponent {
    constructor(mfTable) {
        this.mfTable = mfTable;
        this.isSortedByMeAsc = false;
        this.isSortedByMeDesc = false;
    }
    ngOnInit() {
        this.mfTable.onSortChange.subscribe((event) => {
            this.isSortedByMeAsc = (event.sortBy == this.sortBy && event.sortOrder == "asc");
            this.isSortedByMeDesc = (event.sortBy == this.sortBy && event.sortOrder == "desc");
        });
    }
    sort() {
        if (this.isSortedByMeAsc) {
            this.mfTable.setSort(this.sortBy, "desc");
        }
        else {
            this.mfTable.setSort(this.sortBy, "asc");
        }
    }
};
tslib_1.__decorate([
    Input("by"),
    tslib_1.__metadata("design:type", String)
], DataTableSorterComponent.prototype, "sortBy", void 0);
DataTableSorterComponent = tslib_1.__decorate([
    Component({
        selector: "mfDefaultSorter",
        templateUrl: './datasorter.component.html'
    }),
    tslib_1.__metadata("design:paramtypes", [DataTableService])
], DataTableSorterComponent);
export { DataTableSorterComponent };
